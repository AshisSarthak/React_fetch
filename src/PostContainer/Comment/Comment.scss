.comment-author {
    display: flex;
    align-items: center;
    justify-content: space-between;
}
.comment-cont {
    padding-left: 20px;
    border-bottom: 1px solid #eee;
    padding-bottom: 10px;

    .body {
        color: #666;
    }
}