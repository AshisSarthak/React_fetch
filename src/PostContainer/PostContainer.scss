.post-container {
    width: 60%;
    max-height: 400px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: 1px 0px 2px #aaa;
    overflow: auto;
    margin: 2% 20%;
}
.post-header, .commentSection{
    width: 90%;
    margin-left: 5%;
    border-bottom: 1px solid #eee; 
}
.author {
    display: flex;
    align-items: center;

    h5 {
        margin-left: 15px;
    }
}
