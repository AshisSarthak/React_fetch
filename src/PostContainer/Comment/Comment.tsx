import React from 'react';
import './Comment.scss';
import { PostComment } from '../PostContainer';

export const Comment = (props:{item:  PostComment}) => {
    return (
        <div className='comment-cont'>
            <div className='comment-author'>
                <h5 className="subject">
                    {props.item.name}
                </h5>
                <div className='author-name'>
                    {props.item.email}
                </div>
            </div>
            <div className="body">
                {props.item.name}
            </div>
        </div>
    )
}