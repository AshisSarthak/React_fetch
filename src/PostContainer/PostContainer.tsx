import React from 'react';
import { PostItem } from '../PostList/PostList';
import './PostContainer.scss';
import { UserDetails } from '../User/User';
import { Comment } from './Comment/Comment';

interface PostContainerProp {
  post: PostItem;
}

export interface PostComment {
  postid: number;
  id: number;
  name: string;
  email: string;
  body: string; 
}
interface PostContainerState {
  comments?: PostComment[];
  createdBy?: UserDetails;
}

export class PostContainer extends React.Component<PostContainerProp, PostContainerState> {
  constructor(props: PostContainerProp) {
    super(props);
    this.state = {
      comments: [],
      createdBy: undefined
    }
  }

  componentDidMount(){
    fetch(`https://jsonplaceholder.typicode.com/posts/${this.props.post.id}/comments`)
    .then(response => response.json())
    .then(json => this.setState({
      comments: json,
    }));
    fetch(`https://jsonplaceholder.typicode.com/users/${this.props.post.userId}`)
    .then(response => response.json())
    .then(json => this.setState({
      createdBy: json,
    }));
  }

  render(){
    return (
      <div className="post-container">
        <div className="post-header">
          <h4>{this.props.post.title}</h4>
          <div className="author">
            <sub>Created By </sub> <h5>{this.state.createdBy && this.state.createdBy.name}</h5> 
          </div>
        </div>
        <div className="commentSection">
          {
            this.state.comments && this.state.comments.map((val, index) => {
              return <Comment item={val} key={index} />
            })
          }
        </div>

      </div>
   );  
  }
}

export default PostContainer;
