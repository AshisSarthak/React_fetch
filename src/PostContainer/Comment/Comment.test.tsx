import React from "react";
import { shallow } from "enzyme";
import { Comment } from "./Comment";
import { PostComment } from "../PostContainer";

export const demoComment: PostComment = {
  body: "demo",
  email: "abc@demo.com",
  id: 1,
  name: "demo",
  postid: 11,
};

describe("Comment Componnet", () => {
  it("default render", () => {
    const wrapper = shallow(<Comment item={demoComment} />);
    expect(wrapper).toMatchSnapshot();
  });
});
