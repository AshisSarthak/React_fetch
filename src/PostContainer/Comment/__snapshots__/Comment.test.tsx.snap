// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Componnet default render 1`] = `
<div
  className="comment-cont"
>
  <div
    className="comment-author"
  >
    <h5
      className="subject"
    >
      demo
    </h5>
    <div
      className="author-name"
    >
      abc@demo.com
    </div>
  </div>
  <div
    className="body"
  >
    demo
  </div>
</div>
`;
